File: lib/models/consultation_message.dart
class ConsultationMessage {
  final String id;
  final String consultationId;
  final String senderId;
  final String senderName;
  final String senderRole; // 'doctor' or 'patient'
  final String message;
  final String? attachmentUrl;
  final DateTime timestamp;
  final bool isSystemMessage;

  ConsultationMessage({
    required this.id,
    required this.consultationId,
    required this.senderId,
    required this.senderName,
    required this.senderRole,
    required this.message,
    this.attachmentUrl,
    required this.timestamp,
    this.isSystemMessage = false,
  });

  factory ConsultationMessage.fromJson(Map<String, dynamic> json) {
    return ConsultationMessage(
      id: json['_id'] ?? json['id'] ?? '',
      consultationId: json['consultationId'] ?? '',
      senderId: json['senderId'] ?? '',
      senderName: json['senderName'] ?? '',
      senderRole: json['senderRole'] ?? '',
      message: json['message'] ?? '',
      attachmentUrl: json['attachmentUrl'],
      timestamp: json['timestamp'] != null
          ? DateTime.parse(json['timestamp'])
          : DateTime.now(),
      isSystemMessage: json['isSystemMessage'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'consultationId': consultationId,
      'senderId': senderId,
      'senderName': senderName,
      'senderRole': senderRole,
      'message': message,
      if (attachmentUrl != null) 'attachmentUrl': attachmentUrl,
      'timestamp': timestamp.toIso8601String(),
      'isSystemMessage': isSystemMessage,
    };
  }
}

File: icare-backend/models/ConsultationMessage.js
const mongoose = require('mongoose');

const consultationMessageSchema = new mongoose.Schema({
  consultationId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Consultation',
    required: true
  },
  senderId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  senderName: {
    type: String,
    required: true
  },
  senderRole: {
    type: String,
    enum: ['doctor', 'patient'],
    required: true
  },
  message: {
    type: String,
    required: true
  },
  attachmentUrl: String,
  isSystemMessage: {
    type: Boolean,
    default: false
  },
  timestamp: {
    type: Date,
    default: Date.now
  }
}, {
  timestamps: true
});

// Index for faster queries
consultationMessageSchema.index({ consultationId: 1, timestamp: 1 });

module.exports = mongoose.model('ConsultationMessage', consultationMessageSchema);
